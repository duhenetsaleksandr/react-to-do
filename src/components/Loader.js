import React from 'react';
import '../styles/loader.scss';

export default class Loader extends React.Component {
    render() {
        return (<div className="loader"><div className="lds-dual-ring"/></div>);
    }
}
