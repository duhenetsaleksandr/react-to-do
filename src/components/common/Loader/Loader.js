import React from 'react';
import './loader.scss';

const Loader = () => (<div className="loader"><div className="lds-dual-ring"/></div>);

export default Loader;
