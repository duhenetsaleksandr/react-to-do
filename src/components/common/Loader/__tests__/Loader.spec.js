import React from 'react';
import Loader from '../Loader';

describe('should render Loader component', () => {
    let component;
    beforeEach(() => {
        component = mount(<Loader/>);
    });

    it('should contain .loader wrapper', () => {
        const wrapper = component.find('.loader');
        expect(wrapper).toHaveLength(1);
    });

    it('should contain div.lds-dual-ring', () => {
        const wrapper = component.find('.lds-dual-ring');
        expect(wrapper).toHaveLength(1);
    });

    it('should render Loader component snapshot', () => {
        const component = shallow(<Loader/>);
        expect(component).toMatchSnapshot();
    });
});
