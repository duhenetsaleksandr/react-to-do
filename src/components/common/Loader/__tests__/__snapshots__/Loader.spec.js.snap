// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Loader component should render Loader component snapshot 1`] = `
<div
  className="loader"
>
  <div
    className="lds-dual-ring"
  />
</div>
`;
